bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/handler/MeteoAlerteHandler.java

```diff
@@ -0,0 +1,220 @@
+
+/**
+ * The {@link MeteoAlerteHandler} is responsible for updating channels
+ * and querying the API
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class MeteoAlerteHandler extends BaseThingHandler {
+    private static final String URL = "https://public.opendatasoft.com/api/records/1.0/search/?dataset=risques-meteorologiques-copy&"
+            + "facet=etat_vent&facet=etat_pluie_inondation&facet=etat_orage&facet=etat_inondation&facet=etat_neige&facet=etat_canicule&"
+            + "facet=etat_grand_froid&facet=etat_avalanches&refine.nom_dept=";
+    private static final ArrayList<String> ALERT_LEVELS = new ArrayList<>(
+            Arrays.asList("Vert", "Jaune", "Orange", "Rouge"));
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlerteHandler.class);
+    private final Gson gson = new GsonBuilder()
+            .registerTypeAdapter(ZonedDateTime.class, (JsonDeserializer<ZonedDateTime>) (json, type,
+                    jsonDeserializationContext) -> ZonedDateTime.parse(json.getAsJsonPrimitive().getAsString()))
+            .create();
+
+    // Time zone provider representing time zone configured in openHAB configuration
+    private final TimeZoneProvider timeZoneProvider;
+
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+    private @NonNullByDefault({}) String queryUrl;
+
+    public MeteoAlerteHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Météo Alerte handler.");
+
+        MeteoAlerteConfiguration config = getConfigAs(MeteoAlerteConfiguration.class);
+        logger.debug("config department = {}", config.department);
+        logger.debug("config refresh = {}", config.refresh);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        queryUrl = URL + config.department;
+        refreshJob = scheduler.scheduleWithFixedDelay(this::updateAndPublish, 0, config.refresh, TimeUnit.HOURS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Météo Alerte handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAndPublish();
+        } else {
+            logger.debug("The Meteo Alerte binding is read-only and can not handle command {}", command);
+        }
+    }
+
+    private void updateAndPublish() {
+        try {
+            URL url = new URL(queryUrl);
+            try {
+                URLConnection connection = url.openConnection();
+                String response = IOUtils.toString(connection.getInputStream());
+                IOUtils.closeQuietly(connection.getInputStream());
```

Comment:
Any reason you aren't using an HttpClient?

Reply:
Yes, a simple call of this kind should be ok:
String response = HttpUtil.executeUrl("GET", queryUrl, timeout);